custom_components/audiconnect/audi_connect_account.py

```diff
@@ -589,15 +600,22 @@ async def update_vehicle_charger(self):
                 self._vehicle.state["chargingState"] = get_attr(
                     result, "charger.status.chargingStatusData.chargingState.content"
                 )
-                self._vehicle.state["actualChargeRate"] = get_attr(
+                self._vehicle.state["actualChargeRate"] = float(get_attr(

```

Comment:
This fails for my car. In case there is no charge rate available it will be `None `and the float conversion will fail. Error will be `float() argument must be a string or a number, not 'NoneType'`.

Not sure whether my car supports it at all (A6 PHEV), but at least when it's not charging it is `None`. Furthermore down below you already do a parse float: https://github.com/arjenvrh/audi_connect_ha/blob/db71c2243b3713fe1edb7e1a6ae9cac9e74ec975/custom_components/audiconnect/audi_connect_account.py#L1094
Note sure why it's needed here again (just with the /10).

If I have time later I try to find a public charging spot to see whether this works during charging.

Reply:
Just checked when charging and my PHEV does not support this data, hence you need to guard this against `None`.

Reply:
I added a fix for this